ngi_reports/utils/entities.py

```diff
@@ -347,6 +363,50 @@ def populate(self, log, organism_names, **kwargs):
                                         'ApplicationVersion': fc_runp.get('ApplicationVersion')
                                         }
 
+            ## collect info of samples and their library prep / LIMS indexes on the FC (only if -b option is set)
+            if kwargs.get('barcode_from_fc'):
+                log.info('\'barcodes_from_fc\' option was given so index sequences for the report will be taken from the flowcell instead of LIMS')
+                preps_samples_on_fc = []
+                list_additional_samples = []
+
+                ## get all samples from flow cell that belong to the project
+                fc_samples = []
+                for fc_sample in fc_details.get('samplesheet_csv'):
+                    if fc_sample.get('Sample_Name').split('_')[0] == self.ngi_id:
```

Comment:
Would there be samples from other projects in the sample sheet?

Reply:
yes, the sample sheet should contain all samples on the FC. If we have mixed runs, there should be other projects samples on there too. Or do I misunderstand something?